src/pages/index.js

```diff
@@ -78,55 +187,63 @@ function handleClickOutsideModal(e) {
   if (e.target.classList.contains("modal")) closePopup(e.target);
 }
 
-function renderCard(item, method = "prepend") {
-  const cardElement = createCardElement(item);
-  method === "prepend" ? cardList.prepend(cardElement) : cardList.appendChild(cardElement);
-}
-
-function renderInitialCards() {
-  initialCards.forEach((cardData) => renderCard(cardData, "appendChild"));
-}
-
-function addNewCard(name, link) {
-  if (name && link) renderCard({ name, link });
-}
-
-function populateProfileForm() {
-  nameInput.value = profileNameElement.textContent;
-  jobInput.value = profileJobElement.textContent;
-}
-
-function handleFormSubmit(e) {
-  e.preventDefault();
-  e.target.reset();
-  resetValidation(e.target, Settings);
-  closePopup(e.target.closest(".modal"));
-}
-
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileNameElement.textContent = nameInput.value.trim();
-  profileJobElement.textContent = jobInput.value.trim();
-  handleFormSubmit(evt);
+  const submitBtn = editProfileForm.querySelector(".modal__submit-btn");
+  const originalText = submitBtn.textContent;
+  submitBtn.textContent = "Saving...";
+
+  api.editUserInfo({
+    name: nameInput.value,
+    about: jobInput.value
+  })
+    .then(userData => {
+      profileNameElement.textContent = userData.name;
+      profileJobElement.textContent = userData.about;
+      closePopup(editProfileModal);
+    })
+    .catch(err => console.error("Error updating profile:", err))
+    .finally(() => {
+      submitBtn.textContent = originalText;
+    });
 }
 
 function handleCardFormSubmit(evt) {
   evt.preventDefault();
-  addNewCard(captionInput.value.trim(), imageUrlInput.value.trim());
-  handleFormSubmit(evt);
+  const submitBtn = newPostForm.querySelector(".modal__submit-btn");
+  const originalText = submitBtn.textContent;
+  submitBtn.textContent = "Saving...";
+
+  const name = captionInput.value.trim();
+  const link = imageUrlInput.value.trim();
+  api.addNewCard({ name, link })
+    .then(cardData => {
+      const cardElement = createCardElement(cardData);
+      cardList.prepend(cardElement);
+      closePopup(newPostModal);
+      newPostForm.reset();
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.